exercises/dominik_arominski/ex01_basics/src/main.cpp

```diff
@@ -0,0 +1,82 @@
+#include <fstream>
+#include <string>
+#include <map>
+#include <vector>
+#include <algorithm>
+
+std::string RemovePunctuation(std::string word) {
+  const char punctuation[] = {'.', ',', '?', '!', '"', '\'', ')', '('};
+  for (auto punc : punctuation) {
+    if(word.find(punc) < 100) {
+      word = word.substr(0, word.find(punc)) +  word.substr(word.find(punc)+1, word.size());
+    }
+  }
+  return word;
+}
+
+std::vector<std::string> RemoveHyphenation(std::string const word) {
+  std::string const word1 = word.substr(0, word.find("-"));
+  std::string const word2 = word.substr(word.find("-")+1, word.size());
+  return {word1, word2}; 
+}
+
+int main(int argn, char **argc)
+{
+  if (argn < 2) {
+    throw std::runtime_error("File name not provided.");
+  }
+  else if (argn > 2) {
+    throw std::runtime_error("More than one input file has been provided. Please provide only one file name.");
+  }
+  
+  std::fstream inputFile;
+  inputFile.open(argc[1], std::fstream::in);
+  if (!inputFile) {
+    throw std::runtime_error("Could not open input file.");
+  }
+
+  std::map<std::string, int> wordMap;
+  std::string word;
+  while(inputFile >> word) {
+    if (word.size() < 4 ) {
+      continue;
+    }
+    word = RemovePunctuation(word);
+    if (word.find("-") < 100) {
```

Comment:
Again the check, is this in the requirements of the problem?

Reply:
The check should be different, as said before: if (word.find("-") != std::string::npos). The check for hyphenation was in the requirements.